### PR TITLE
feat: drain database jobs on shutdown

### PR DESCRIPTION
**Pull request by tustvold**
This makes the Database worker drain wait for its jobs to finish before "completing", this allows cleanly shutting down a Database - which will be important for Database deletion among other things.

In order to test this, I added the ability DummyJob to be spawned with a database name, which comprises the majority of this PR's changes

### CODE REVIEW

File: server/src/database.rs [hunk 1]
```diff
@@ -581,3 +607,63 @@ async fn get_store_bytes(
...
+        );
+
+        // Should have failed to load (this isn't important to the test)
+        let err = database.wait_for_init().await.unwrap_err();
+        assert!(matches!(err.as_ref(), InitError::RulesFetch { .. }));
+
+        // Database should be running
+        assert!(database.join().now_or_never().is_none());
+
+        let db_tracker = application
```
Comment (crepererum) [inline server/src/database.rs:641]: Can we call this `job_tracker`? I'm pretty confused by "database", "database.join" and "db_tracker".

File: server/src/database.rs [hunk 2]
```diff
@@ -581,3 +607,63 @@ async fn get_store_bytes(
...
+        // Expect timeout to not complete
+        tokio::time::timeout(tokio::time::Duration::from_millis(1), database.join())
+            .await
+            .unwrap_err();
+
+        // Database task shouldn't have finished yet
+        assert!(!db_tracker.is_complete());
+
+        // Wait for database to finish
+        database.join.await.unwrap();
```
Comment (crepererum) [inline server/src/database.rs:661]: why is this additional "join" necessary, shouldn't this be done by `database.shutdown`?

Comment (tustvold) [reply server/src/database.rs:661]: Shutdown doesn't block, it just signals shutdown, perhaps it should be named something else?

File: server/src/database.rs (hunk 1 above)
Comment (tustvold) [reply server/src/database.rs]: I renamed it, hopefully it is less confusing now?

File: server/src/database.rs (hunk 2 above)
Comment (crepererum) [reply server/src/database.rs:663]: Then maybe there should be two functions then: `signal_shutdown` and `wait_shutdown_completed`? And then the join handle can be private.

Comment (tustvold) [reply server/src/database.rs:663]: Oh! The join handle is private, there is a member function called join(), this is just accidentally using the member directly - will fix

**Review by crepererum — APPROVED** [review]
Now it's clearer, thanks!

**Pull request merged**